### PR TITLE
Add travis python test

### PR DESCRIPTION
**Pull request by Linary**

### CODE REVIEW

File: hubble-dist/assembly/travis/start-hubble [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+main_class="com.baidu.hugegraph.HugeGraphHubble"
+args=${CONF_PATH}/hugegraph-hubble.properties
+log=${LOG_PATH}/hugegraph-hubble.log
+echo "starting HugeGraphHubble, logging to ${log}"
+
+nohup nice -n 0 java -server ${java_opts} ${agent_opts} -cp ${class_path} ${main_class} ${args} > ${log} 2>&1 < /dev/null &
+echo $! > ${PID_FILE}
+
+# ensure hubble start
+sleep 30
```
Comment (zhoney) [inline hubble-dist/assembly/travis/start-hubble]: Is there more robust method to ensure hubble started?

File: hubble-dist/assembly/travis/hugegraph-server2/rest-server.properties [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# bind url
+restserver.url=http://127.0.0.1:8081
+gremlinserver.url=http://127.0.0.1:8183
+
+# graphs list with pair NAME:CONF_PATH
+graphs=[hugegraph1:conf/hugegraph1.properties]
```
Comment (javeme) [inline hubble-dist/assembly/travis/hugegraph-server2/rest-server.properties]: hugegraph3

File: hubble-dist/assembly/travis/run-test.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
```
Comment (javeme) [inline hubble-dist/assembly/travis/run-test.sh]: rename to run-api-test.sh

File: hubble-dist/assembly/travis/start-hubble [hunk 4]
```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
```
Comment (javeme) [inline hubble-dist/assembly/travis/start-hubble]: start-hubble.sh

File: hubble-dist/assembly/travis/hugegraph-server2/gremlin-server.yaml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+port: 8183
+
+# timeout in ms of gremlin query
+scriptEvaluationTimeout: 30000
+
+# If you want to start gremlin-server for gremlin-console(web-socket),
+# please change `HttpChannelizer` to `WebSocketChannelizer` or comment this line.
+channelizer: org.apache.tinkerpop.gremlin.server.channel.HttpChannelizer
+graphs: {
+  hugegraph1: conf/hugegraph1.properties
```
Comment (javeme) [inline hubble-dist/assembly/travis/hugegraph-server2/gremlin-server.yaml]: prefer hugegraph3

File: hubble-dist/assembly/travis/start-hubble (hunk 1 above)
Comment (Linary) [reply hubble-dist/assembly/travis/start-hubble]: use wait_for_startup

File: hubble-dist/assembly/travis/start-hubble.sh [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        ;;
+    esac
+    shift
+done
+
+if [[ -f ${PID_FILE} ]] ; then
+    pid=`cat ${PID_FILE}`
+    if kill -0 ${pid} > /dev/null 2>&1; then
+        echo "HugeGraphHubble is running as process ${pid}, please stop it first!"
+        exit
```
Comment (javeme) [inline hubble-dist/assembly/travis/start-hubble.sh]: exit 1

File: hubble-dist/assembly/travis/start-hubble.sh [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+for jar in `ls ${LIB_PATH}/*.jar`; do
+    class_path=${class_path}:${jar}
+done
+
+java_opts="-Xms512m"
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        --help|-help|-h)
+        print_usage
+        exit
```
Comment (javeme) [inline hubble-dist/assembly/travis/start-hubble.sh]: exit 1

File: hubble-be/src/test/python/steps/check_server_status.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    http_url = "http://" + url + "/api/v1.1/actuator/health"
+    res = requests.get(http_url)
+    context.res_json = res.json()
+
+@then("response:(?P<response>.+)")
+def step_impl(context, response):
+    res_json = context.res_json
+    ass_json = json.loads(response)
+
+    if res_json["status"] == ass_json["status"]:
```
Comment (javeme) [inline hubble-be/src/test/python/steps/check_server_status.py]: use assertEqual() instead

File: hubble-be/src/test/python/steps/check_server_status.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+import requests
+import sys
+import json
+
+reload(sys)
+sys.setdefaultencoding('utf8')
+use_step_matcher("re")
+
+@when("scene:(?P<scene>.+) url:(?P<url>.+)")
+def step_impl(context, scene, url):
```
Comment (javeme) [inline hubble-be/src/test/python/steps/check_server_status.py]: can we define a base class?

File: hubble-be/src/test/python/steps/check_server_status.py (hunk 8 above)
Comment (Linary) [reply hubble-be/src/test/python/steps/check_server_status.py]: used assert

File: hubble-be/src/test/python/steps/check_server_status.py (hunk 9 above)
Comment (Linary) [reply hubble-be/src/test/python/steps/check_server_status.py]: seems a little complicated

File: hubble-dist/assembly/travis/start-hugegraph.sh [hunk 10]
```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
```
Comment (javeme) [inline hubble-dist/assembly/travis/start-hugegraph.sh]: remove this shell files copied from server

File: hubble-be/src/test/python/steps/check_server_status.py [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    http_url = "http://" + url + "/api/v1.1/actuator/health"
+    res = requests.get(http_url)
+    context.res_json = res.json()
+
+@then("response:(?P<response>.+)")
+def step_impl(context, response):
+    res_json = context.res_json
+    ass_json = json.loads(response)
+
+    assert ass_json["status"] == res_json["status"]
```
Comment (javeme) [inline hubble-be/src/test/python/steps/check_server_status.py]: use assertpy instead

File: .travis.yml [hunk 12]
```diff
@@ -16,17 +18,25 @@ branches:
...
-  - $TRAVIS_DIR/install-hugegraph.sh $TRAVIS_BRANCH | grep -v "Downloading\|Downloaded"
+  - $TRAVIS_DIR/download-hugegraph.sh $SERVER_VERSION
+  - $TRAVIS_DIR/hugegraph-server1/install-hugegraph.sh
+  - $TRAVIS_DIR/hugegraph-server2/install-hugegraph.sh
+  - $TRAVIS_DIR/install-hugegraph-hubble.sh
+  - jps
 
 script:
-  - mvn test -Punit
+  - $TRAVIS_DIR/run-api-test.sh
```
Comment (javeme) [inline .travis.yml]: keep unit test

File: hubble-dist/assembly/travis/run-api-test.sh [hunk 13]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -ev
+
+TRAVIS_DIR=`dirname $0`
+
+behave hubble-be/src/test/python
```
Comment (javeme) [inline hubble-dist/assembly/travis/run-api-test.sh]: $TRAVIS_DIR/hubble-be/src/test/python ?

Comment (Linary) [reply hubble-dist/assembly/travis/run-api-test.sh]: No, just hubble-be/src/test/python, $TRAVIS_DIR is hubble-dist/assembly/travis/xxx

File: hubble-dist/assembly/travis/start-hubble.sh [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+BIN_PATH=${HOME_PATH}/bin
+CONF_PATH=${HOME_PATH}/conf
+LIB_PATH=${HOME_PATH}/lib
+LOG_PATH=${HOME_PATH}/logs
+PID_FILE=${BIN_PATH}/pid
+
+. ${BIN_PATH}/common_functions
+
+print_usage() {
+    echo "  usage: start-hubble [options]"
```
Comment (zhoney) [inline hubble-dist/assembly/travis/start-hubble.sh]: usage: start-hubble.sh [options]

File: hubble-dist/assembly/travis/start-hubble.sh [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+
+class_path="."
+for jar in `ls ${LIB_PATH}/*.jar`; do
+    class_path=${class_path}:${jar}
+done
+
+java_opts="-Xms512m"
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        --help|-help|-h)
```
Comment (zhoney) [inline hubble-dist/assembly/travis/start-hubble.sh]: keep consistency with help info

File: hubble-dist/assembly/travis/start-hubble.sh [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+        print_usage
+        exit 1
```
Comment (zhoney) [inline hubble-dist/assembly/travis/start-hubble.sh]: exit 0

File: hubble-dist/assembly/travis/hugegraph-server2/gremlin-server.yaml [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+scriptEngines: {
+  gremlin-groovy: {
+    plugins: {
+      com.baidu.hugegraph.plugin.HugeGraphGremlinPlugin: {},
+      org.apache.tinkerpop.gremlin.server.jsr223.GremlinServerGremlinPlugin: {},
+      org.apache.tinkerpop.gremlin.jsr223.ImportGremlinPlugin: {
+        classImports: [
+          java.lang.Math,
+          com.baidu.hugegraph.util.DateUtil,
+          com.baidu.hugegraph.traversal.optimize.Text
```
Comment (zhoney) [inline hubble-dist/assembly/travis/hugegraph-server2/gremlin-server.yaml]: add TraverserUtil

**Pull request merged**